pkg/syncer/fullsync.go

```diff
@@ -467,15 +497,54 @@ func volumeInfoCRFullSync(ctx context.Context, metadataSyncer *metadataSyncInfor
 					// Add volume to deletion map.
 					volumeInfoCrDeletionMap[vc][cnsvolumeinfo.Spec.VolumeID] = true
 				}
+			} else {
+				// get aggregated snapshotsize from cns and update cnsvolumeinfo
+				if isStorageQuotaM2FSSEnabled {
+					if cnsVol, ok := cnsVolumeMap[cnsvolumeinfo.Spec.VolumeID]; ok {
+						updateVolumeInfoWithSnapshotDetails(ctx, cnsvolumeinfo, cnsVol)
+					}
+				}
 			}
 		}
 	}
 	log.Debugf("FullSync for VC %s: volumeInfoCrDeletionMap: %v", vc, volumeInfoCrDeletionMap)
 }
 
+// updateVolumeInfoWithSnapshotDetails update cnsvolumeinfo with snapshotdetails by comparing the aggregatedSnapshotSize of volume.
+func updateVolumeInfoWithSnapshotDetails(ctx context.Context, cnsvolumeinfo *cnsvolumeinfov1alpha1.CNSVolumeInfo, cnsVol cnstypes.CnsVolume) error {
+	log := logger.GetLogger(ctx)
+	log.Info("Validate Snapshot details for volumeID %q", cnsvolumeinfo.Spec.VolumeID)
+	var aggregatedSnapshotCapacity int64
+	val, ok := cnsVol.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
+	if ok {
+		aggregatedSnapshotCapacity = val.AggregatedSnapshotCapacityInMb
+	}
+	if aggregatedSnapshotCapacity != cnsvolumeinfo.Spec.AggregatedSnapshotSize.Value() {
```

Comment:
We have to be careful not to set the aggregated snapshot size to "-1." Please ensure that the minimum aggregated snapshot size is 0.

Reply:
it will not set size to -1, in func call GetValidatedCNSVolumeInfoPatch, if size is -1 we are not updating aggregatedSnapshotCapacity. and set only validaggregatedsnapshotsize to false